packages/components/src/components/atoms/form/RequestTypeItem.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useMemo } from 'react'
+import { styled } from '@mui/system'
+import TreeSelector from './TreeSelector'
+
+import {
+  ICategoryLimitations,
+  ILimitationsTypes,
+  IOptionsTags,
+  IRequestType,
+  IRequestTypesOptions,
+  ISearchLimitations,
+  ITreeItem,
+  flatTree,
+} from '@elastic-suite/gally-admin-shared'
+
+import TextFieldTagsMultiple from './TextFieldTagsMultiple'
+
+import Checkbox from './Checkbox'
+import { IconButton } from '@mui/material'
+import IonIcon from '../IonIcon/IonIcon'
+import {
+  CustomFirstSelectedItem,
+  CustomRootItem,
+  CustomSelectedItem,
+} from './RequestTypeItem.styled'
+
+const CustomItem = styled('div')(({ theme }) => ({
+  display: 'flex',
+  gap: theme.spacing(3),
+  alignItems: 'start',
+}))
+
+const CustomLabel = styled('div')(({ theme }) => ({
+  color: theme.palette.colors.neutral[900],
+  fontFamily: 'var(--gally-font)',
+  fontSize: theme.spacing(1.5),
+  fontWeight: '600',
+  width: '190px',
+  lineHeight: '18px',
+}))
+
+const CustomDataLimitations = styled('div')(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'row',
+  gap: theme.spacing(2),
+  alignItems: 'center',
+}))
+
+export interface IRequestTypeItem {
+  value: IRequestType
+  onChange: (value: IRequestType) => void
+  options: IOptionsTags[]
+  limitationsTypes: ILimitationsTypes[]
+  requestTypesOptions: IRequestTypesOptions[]
+  categoriesList: ITreeItem[]
+}
+
+interface IPropsComponent {
+  disabled?: boolean
+  onChange: (data: ITreeItem[] | ISearchLimitations[]) => void
+}
+
+function RequestTypeItem(props: IRequestTypeItem): JSX.Element {
+  const {
+    value,
+    onChange,
+    options,
+    limitationsTypes,
+    requestTypesOptions,
+    categoriesList,
+  } = props
+
+  function onChangeApplyToAll(
+    idItem: IRequestTypesOptions[],
+    val: boolean
+  ): void {
+    const newData = value.requestTypes.map((item) => {
+      if (
+        item.requestType ===
+        (idItem as IRequestTypesOptions[]).find(
+          (it) => it.value === item.requestType
+        )?.value
```

Comment:
Pour éviter de faire un `find` tu peux créer une liste via un `map` de tout les `it.value` et faire un `includes` c'est plus performant et plus lisible aussi.

En dehors de la boucle:
```js
const requestTypeOptionValues = (idItem as IRequestTypesOptions[]).map((it) => it.value)
```

Dans la boucle:
```js
if (requestTypeOptionValues.includes(item.requestType)) {
```